test/System.IdentityModel.Tokens.Jwt.Tests/JwtSecurityTokenHandlerTests.cs

```diff
@@ -463,6 +464,99 @@ public void InboundOutboundClaimTypeMapping()
             RunClaimMappingVariation(jwt, handler, validationParameters, expectedClaims: expectedClaims, identityName: null);
         }
 
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+        [Theory, MemberData("ReadJwtTokenTheoryData")]
+#pragma warning restore CS3016 // Arrays as attribute arguments is not CLS-compliant
+        public void ReadJwtToken(JwtTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ReadJwtToken", theoryData);
+            var context = new CompareContext($"{this}.ReadJwtToken, {theoryData.TestId}");
+            try
+            {
+                var principal = theoryData.TokenHandler.ValidateToken(theoryData.Token, theoryData.ValidationParameters, out SecurityToken validToken);
+
+                var jwtToken1 = validToken as JwtSecurityToken;
+                var jwtToken2 = theoryData.SecurityToken as JwtSecurityToken;
+
+                string time1 = Convert.ToString(jwtToken1.Payload[JwtRegisteredClaimNames.Nbf], CultureInfo.InvariantCulture);
+                string time2 = Convert.ToString(jwtToken2.Payload[JwtRegisteredClaimNames.Nbf], CultureInfo.InvariantCulture);
+                IdentityComparer.AreEqual(time1, time2, context);
+
+                time1 = Convert.ToString(jwtToken1.Payload[JwtRegisteredClaimNames.Exp], CultureInfo.InvariantCulture);
+                time2 = Convert.ToString(jwtToken2.Payload[JwtRegisteredClaimNames.Exp], CultureInfo.InvariantCulture);
+                IdentityComparer.AreEqual(time1, time2, context);
+
+                time1 = Convert.ToString(jwtToken1.Payload[JwtRegisteredClaimNames.Iat], CultureInfo.InvariantCulture);
+                time2 = Convert.ToString(jwtToken2.Payload[JwtRegisteredClaimNames.Iat], CultureInfo.InvariantCulture);
+                IdentityComparer.AreEqual(time1, time2, context);
+
+                theoryData.ExpectedException.ProcessNoException(context);
+            }
+            catch (Exception ex)
+            {
+                theoryData.ExpectedException.ProcessException(ex, context);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<JwtTheoryData> ReadJwtTokenTheoryData
+        {
+            get
+            {
+                TimeSpan timeSpan = DateTime.UtcNow.ToUniversalTime().AddMinutes(10) - EpochTime.UnixEpoch;
+                JwtSecurityTokenHandler handler = new JwtSecurityTokenHandler();
+
+                JwtSecurityToken jwtToken1 = handler.CreateToken(new SecurityTokenDescriptor()) as JwtSecurityToken;
+                jwtToken1.Payload[JwtRegisteredClaimNames.Nbf] = 1471924298.67746;
+                jwtToken1.Payload[JwtRegisteredClaimNames.Iat] = jwtToken1.Payload[JwtRegisteredClaimNames.Nbf];
```

Comment:
why not just assign the value as above '= 1471924298.67746'

Reply:
Agree. Same as above.